src/hyperion/experiment_plans/grid_detect_then_xray_centre_plan.py

```diff
@@ -161,11 +189,17 @@ def run_grid_detection_plan(
     )
     parameters.hyperion_params.ispyb_params.upper_left = out_upper_left
 
-    grid_params = grid_params_callback.get_grid_parameters()
+    if parameters.experiment_params.use_panda:
+        grid_params = grid_params_callback.get_panda_grid_parameters()
 
-    flyscan_xray_centre_parameters = create_parameters_for_flyscan_xray_centre(
-        parameters, grid_params
-    )
+        flyscan_xray_centre_parameters = (
+            create_parameters_for_panda_flyscan_xray_centre(parameters, grid_params)
+        )
+    else:
+        grid_params = grid_params_callback.get_grid_parameters()
+        flyscan_xray_centre_parameters = create_parameters_for_flyscan_xray_centre(
+            parameters, grid_params
+        )
```

Comment:
Nit: There's no reason you can't make these parameters later and just have the one `if` statement for `use_panda`